src/handlers/QuotingHandler.js

```diff
@@ -0,0 +1,192 @@
+/* eslint-disable space-before-function-paren */
+const { AuditEventAction } = require('@mojaloop/event-sdk')
+const { Enum } = require('@mojaloop/central-services-shared')
+const { reformatFSPIOPError } = require('@mojaloop/central-services-error-handling').Factory
+
+const { ErrorMessages } = require('../lib/enum')
+const { getSpanTags } = require('../lib/util')
+const dto = require('../lib/dto')
+
+const { FSPIOP } = Enum.Http.Headers
+
+class QuotingHandler {
+  constructor (deps) {
+    this.quotesModelFactory = deps.quotesModelFactory
+    this.bulkQuotesModelFactory = deps.bulkQuotesModelFactory
+    this.logger = deps.logger
+    this.config = deps.config
+    this.cache = deps.cache
+    this.tracer = deps.tracer
+    this.handleMessages = this.handleMessages.bind(this)
+  }
+
+  async handleMessages(error, messages) {
+    // think, if we need to add Metrics.getHistogram here
+    if (error) {
+      this.logger.error(`${ErrorMessages.consumingErrorFromKafka}: ${error.message}`, { error })
+      throw reformatFSPIOPError(error)
+    }
+
+    const results = await Promise.allSettled(
+      messages.map(msg => this.defineHandlerByTopic(msg))
+    )
+    this.logger.info('handleMessages results:', results)
+
+    return true
+  }
+
+  async defineHandlerByTopic(message) {
+    const { topic, requestData } = dto.requestDataFromMessageDto(message)
+    const { QUOTE, BULK_QUOTE } = this.config.kafkaConfig.CONSUMER
+
+    switch (topic) {
+      case QUOTE.POST.topic:
+        return this.handlePostQuotes(requestData)
+      case QUOTE.PUT.topic:
+        return this.handlePutQuotes(requestData)
+      case QUOTE.GET.topic:
+        return this.handleGetQuotes(requestData)
+      case BULK_QUOTE.POST.topic:
+        return this.handlePostBulkQuotes(requestData)
+      case BULK_QUOTE.PUT.topic:
+        return this.handlePutBulkQuotes(requestData)
+      case BULK_QUOTE.GET.topic:
+        return this.handleGetBulkQuotes(requestData)
+
```

Comment:
As far as we agreed to publish POST/PUT/GET quotes/bulkQuotes payloads to a separate queues, what's wrong with having a separate processing function per queue?  Do you think, in topic `topic-quotes-post` we might have a message with type `bulkQueue` and action `get`?

Reply:
No, that's not what I am proposing. We can have separate function for each queue. I am just saying, instead of comparing the topic name, we can have event / action to determine the type of message like we are doing in central-ledger.

Reply:
Oh wait, I think you are right. I don't see any reason to determine the message based on the topic name. I think its fine to have this logic here. But lets keep the event and action metadata in the kafka message just incase. You might already doing that, but just a reminder.